types.c

```diff
@@ -48,7 +48,7 @@ fltype_t *get_type(value_t t)
             ft->elsz = eltype->size;
             ft->eltype = eltype;
             ft->init = &cvalue_array_init;
-            eltype->artype = ft;
+            //eltype->artype = ft; -- this is a bad idea since some types carry array sizes
```

Comment:
Maybe... it would be safe to do that. It seems that `(array <type> ...)` calls, and constructors like `#vu8()` and string construction with quotes all use `get_array_type()`. Only `(c-value ...)` seems to use `(get-type ...)` directly for array construction. So, it seems unnecessary to set `artype` here, but I suppose it could be suboptimal under some unusual scenarios.